app/services/authorization/user_permissible_service.rb

```diff
@@ -0,0 +1,138 @@
+module Authorization
+  class UserPermissibleService
+    attr_accessor :user
+
+    def initialize(user)
+      @user = user
+      @cached_permissions = Hash.new do |hash, key|
+        hash[key] = Set.new
+      end
```

Comment:
If the checks for the current permission turn out to be more complicated (e.g. because of Non member and Anonymous), it would be an option to use a utility class inheriting from `Hash` that will do the fetching internally. Part of the implementation in this class could be moved into the utility class: 
* https://github.com/opf/openproject/pull/13809/files#diff-2c66bd35b7924325e768d995449b664d8a7828da69cde64deab84919bf3a1777R112-R136
* https://github.com/opf/openproject/pull/13809/files#diff-2c66bd35b7924325e768d995449b664d8a7828da69cde64deab84919bf3a1777R68-R74